core/Plugin/Manager.php

```diff
@@ -319,6 +319,10 @@ public function isPluginLoaded($name)
      */
     public function readPluginsDirectory()
     {
+        if (!empty($GLOBALS['MATOMO_PLUGIN_NAMES_AVAILABLE'])) {
```

Comment:
this along with the other related changes saves 1-2% of each tracking request. Eventually we could add more "caching" around this.

Reply:
In `$matomoDir/bootstrap.php` someone could basically configure which plugin names are available in an array.